zoo_frontend/src/util/withAuth.jsx

```diff
@@ -41,17 +39,18 @@ export default (allowedRoles = ['authenticated']) => (WrappedComponent) => {
         return { ...pageProps };
       }
 
-      const api = new Api(c.authToken || '');
-
+      const api = new Api(c.authToken);
       try {
-        await api.validateToken().catch(() => {
+        await api.validateToken().then(async () => {
+          // console.log('pass validation');
+        }, () => {
           if (process.browser) {
             document.cookie = 'authToken=; path=/';
           }
           api.setToken('');
-          return {};
+          redirectTo('/login', { res: ctx.res, status: 301 });
+          return { ...pageProps };
```

Comment:
I think this is the part that it is messing up. Returning from here does nothing. To do the return correctly you would have to use a try/catch